src/capturer/engine/win/mod.rs

```diff
@@ -21,8 +21,14 @@ struct Capturer {
     pub crop: Option<Area>,
 }
 
-pub struct WinStream {
-    settings: Settings<FlagStruct, WCMonitor>,
+#[derive(Clone)]
+enum Settings {
+    Window(WCSettings<FlagStruct, WCWindow>),
+    Display(WCSettings<FlagStruct, WCMonitor>),
```

Comment:
Settings having these values feels weird.
What if we decide to add more parameters to the settings in the future. How would that work?

Do you think maybe we can have a `targets` field inside Settings which contains `Window` & `Display` ?

Reply:
That is what I tried first 😭

Unfortunately it didn't work well because the windows-capture `Settings` struct (`WCSettings` here) expects to get a flag that needs specific`TryFrom` and `TryInto` impl. Trying to make that led a lot of duplicated code on our side and importing many error types to make it work so we gave up on it.